File: Demos/Tutorials/Groups/Delphi/GroupsTutUnit1.pas
unit GroupsTutUnit1;

interface

{$I vtk.inc}

uses
  Windows, Messages, SysUtils, {$IFDEF VTK_D6_OR_D7} Variants, {$ENDIF} Classes, Graphics, Controls, Forms,
  Dialogs, DB, DBTables, vgr_WorkbookDesigner, vgr_Report,
  vgr_CommonClasses, vgr_DataStorage, StdCtrls;

type
  TForm1 = class(TForm)
    vgrWorkbook1: TvgrWorkbook;
    vgrReportEngine1: TvgrReportEngine;
    vgrWorkbookDesigner1: TvgrWorkbookDesigner;
    Query1: TQuery;
    Button1: TButton;
    vgrReportTemplate1: TvgrReportTemplate;
    vgrReportTemplate1vgrReportTemplateWorksheet1: TvgrReportTemplateWorksheet;
    vgrDetailBand1: TvgrDetailBand;
    vgrBand1: TvgrBand;
    vgrGroupBand1: TvgrGroupBand;
    vgrBand2: TvgrBand;
    vgrDataBand1: TvgrDataBand;
    vgrBand3: TvgrBand;
    procedure Button1Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.Button1Click(Sender: TObject);
begin
  vgrReportEngine1.Generate;
  vgrWorkbookDesigner1.Design(True);
end;

end.

File: Demos/Tutorials/Groups/CBuilder/GroupsTutUnit1.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "GroupsTutUnit1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "vgr_CommonClasses"
#pragma link "vgr_DataStorage"
#pragma link "vgr_Report"
#pragma link "vgr_WorkbookDesigner"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button1Click(TObject *Sender)
{
  vgrReportEngine1->Generate();
  vgrWorkbookDesigner1->Design(TRUE);        
}
//---------------------------------------------------------------------------
